hydra/redisgraph_demo1.py

```diff
@@ -0,0 +1,258 @@
+            # get_function(node) which shows the properties of node
+        elif new_obj_op["method"] == "POST":
+            post_endpoint = 1
+            return_class = new_obj_op["returns"]
+            class_node = Node(
+                label="id",
+                alias="classes",
+                properties=node.properties)
+            redis_graph.add_node(class_node)
+            edge = Edge(node, "post_method", class_node)
+            redis_graph.add_edge(edge)
+            # here we'll call a function for post the endpoint
+        elif new_obj_op["method"] == "PUT":
+            put_endpoint = 1
+            return_class = new_obj_op["returns"]
+            class_node = Node(
+                label="id",
+                alias="classes",
+                properties=node.properties)
+            redis_graph.add_node(class_node)
+            edge = Edge(node, "put_method", class_node)
+            redis_graph.add_edge(edge)
+    if post_endpoint == 0 and put_endpoint == 0:
+        flag = 0
+        for obj in apidoc["supportedClass"]:
+            if obj["@id"] == returns:
+                flag = 1
+                collection = obj
+        if flag == 1:
+            for obj in collection["supportedOperation"]:
+                if obj["method"] == "POST" or obj["method"] == "PUT":
+                    return_class = obj["returns"]
+                    class_node = Node(
+                        label="id",
+                        alias="classes",
+                        properties=node.properties)
+                    redis_graph.add_node(class_node)
+                    break
+                if obj["method"] == "POST":
+                    post_collection = 1
+                if obj["method"] == "PUT":
+                    put_collection = 1
+
+    """ connect the collection to the classes by edge"""
+
+    edge = Edge(collection_node, "has_classes", class_node)
+    redis_graph.add_edge(edge)
+    if (post_endpoint == 1 or
+            put_endpoint == 1 or
+            post_collection == 1 or
+            put_collection == 1):
+        flag = 0
+        for obj in apidoc["supportedclass"]:
+            if obj["@id"] == return_class:
+                flag = 1
+                classes = obj
+        if flag == 1:
+            for obj in classes["supportedOperation"]:
+                if obj["method"] == "PUT":
+                    put_classes = 1
+                if obj["method"] == "POST":
+                    post_classes = 1
+                if obj["method"] == "DELETE":
+                    delete_classes = 1
+                if obj["method"] == "GET":
+                    get_classes = 1
+
+    """ members are leave nodes which is connect with the classes
+        every elementary operation should be perform on these.
+        like put,post,delete these operation can be done by adding and
+        deleting the nodes and update all the above nodes properties"""
+
+    for member in new_file["members"]:
```

Comment:
Again, not able to understand at all what is happening in the code, the output simply says that some members are added to the Redis graph, please define what the graph structure is and explain how the information is stored in Redis. Also, please write give some proper output or reference so that we can view the graph as well.

Reply:
yeah, here I am tried to add the events(members) as the elementary nodes and store it into Redis. We can update the graph if any other event is posted or update only by adding or deleting the node and update the property in above nodes.